pkg/infra/tower/tower.go

```diff
@@ -3,42 +3,72 @@ package tower
 import (
 	"log"
 	"sync"
-
-	"github.com/divan/goabm/abm"
 )
 
 type Tower struct {
-	FoodOnPlatform  float64
-	FloorOfPlatform uint64
+	currPlatFood    float64
+	maxPlatFood     float64
+	currPlatFloor   uint64
 	mx              sync.RWMutex
-	agents          []abm.Agent
-	AgentCount      int
+	agentCount      int
+	agents          map[string]BaseAgentCore
```

Comment:
I think if you change this to a map of pointers to structs then you can make `setHP()` a lot cleaner i.e. it would become:
```
tower.agents[id].hp = newHP
```

Reply:
As well as most of your other setter functions in tower-agent.go I think

Reply:
TODOed